controllers/storagecluster/cephfilesystem.go

```diff
@@ -47,19 +47,46 @@ func (r *StorageClusterReconciler) newCephFilesystemInstances(initStorageCluster
 	}
 
 	// not in provider mode
+	// standalone deployment that isn't in provider cluster will not have storageProfile
+	// we need to define default dataPool, if storageProfile is set this will be overridden.
 	if !initStorageCluster.Spec.AllowRemoteStorageConsumers {
-		// standalone deployment that isn't in provider cluster will not
-		// have storageProfile, we need to define default dataPool, if
-		// storageProfile is set this will be overridden.
 		ret.Spec.DataPools = []cephv1.NamedPoolSpec{
 			{
-				PoolSpec: cephv1.PoolSpec{
-					DeviceClass:   generateDeviceClass(initStorageCluster),
-					Replicated:    generateCephReplicatedSpec(initStorageCluster, "data"),
-					FailureDomain: initStorageCluster.Status.FailureDomain,
-				},
+				PoolSpec: initStorageCluster.Spec.ManagedResources.CephFilesystems.DataPoolSpec,
 			},
 		}
+		// Append additional pools from specified additional data pool specs
+		for _, dataPoolSpec := range initStorageCluster.Spec.ManagedResources.CephFilesystems.AdditionalDataPoolsSpecs {
+			ret.Spec.DataPools = append(ret.Spec.DataPools, cephv1.NamedPoolSpec{
+				PoolSpec: dataPoolSpec,
+			})
+		}
```

Comment:
The `for _, <value> := range <slice>` for slices containing non-pointers will result in a copy of the item in the slice instead of a pointer to it. And that will happen for each item in the slice. A better approach is:

```suggestion
		specs := &initStorageCluster.Spec.ManagedResources.CephFilesystems.AdditionalDataPoolsSpecs
		for i,  := range specs {
			dataPoolSpec :=  &specs[i]
			ret.Spec.DataPools = append(ret.Spec.DataPools, cephv1.NamedPoolSpec{
				PoolSpec: dataPoolSpec,
			})
		}
```

Which will result in zero-copies of deep structs